src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,41 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    static final int MIN_AGE = 18;
+    private static final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Not valid age: "
+                    + user.getAge() + ". Min allowed age is " + MIN_AGE);
+        }
+        if (isInvalidUser(user)) {
+            throw new RegistrationException("Invalid user data for registration");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists");
+        }
+        return storageDao.add(user);
+    }
+
+    private boolean isInvalidUser(User user) {
+        return user.getLogin().length() < 6
+                || user.getPassword().length() < 6
+                || user.getAge() < 18; // I can't move it to constant field
     }
```

Comment:
why? you should be able to
throw more informative messages, remove this method where you check everything, if login is invalid, inform user, that the problem is in login, for example